nanshe_ipython.ipynb

```diff
@@ -1165,16 +1172,26 @@
         "\n",
         "mskctr_srcs.callback = CustomJS(\n",
         "    args=dict(\n",
+        "        all_tr_dtype_srcs=all_tr_dtype_srcs,\n",
         "        all_tr_shape_srcs=all_tr_shape_srcs,\n",
         "        all_tr_srcs=all_tr_srcs,\n",
         "        tr_srcs=tr_srcs\n",
         "    ), code=\"\"\"\n",
         "    var range = function(n){ return Array.from(Array(n).keys()); };\n",
         "\n",
-        "    var inds = cb_obj.get('selected')['1d'].indices;\n",
-        "    var traces = all_tr_srcs.get('data')['traces'];\n",
+        "    var traces_not_decoded = (all_tr_dtype_srcs.get('data')['traces_dtype'] == 0);\n",
+        "    var traces_dtype = all_tr_dtype_srcs.get('data')['traces_dtype'].constructor;\n",
         "    var traces_shape = all_tr_shape_srcs.get('data')['traces_shape'];\n",
         "    var trace_len = traces_shape[1];\n",
+        "    var traces = all_tr_srcs.get('data')['traces'];\n",
+        "    if (traces_not_decoded) {\n",
+        "        traces = window.pako.inflate(traces);\n",
+        "        traces = new traces_dtype(traces.buffer);\n",
```

Comment:
First, decompress the data with pako. The result will also be a `Uint8Array` just as the input is. Then take the underlying `buffer` of the `Uint8Array` and create a view onto it using a `TypedArray` matching the original type of the data. As we create a view onto the underlying data, we avoid a potentially expensive copy in this step.